### ISSUE TITLE
Include has_many through: relationship fails

### ISSUE THREAD

**Issue by msbit** [JSONAPI-Resources/jsonapi-resources#643]
With the following set of models:

```
class User < ActiveRecord::Base
    ...
    has_many :participations
    has_many :attempts, through: :participations
    has_many :completed_attempts, -> { completed }, through: :participations, source: :attempts
    ...
end
```

```
class Participation < ActiveRecord::Base
    ...
    belongs_to :user
    has_many :attempts
    ...
end
```

```
class Attempt < ActiveRecord::Base
    ...
    belongs_to :participation
    ...
end
```

the request `GET /users/1?include=completed-attempts` fails. [completed-attempts.txt](https://github.com/cerebris/jsonapi-resources/files/152077/completed-attempts.txt).

Likewise, `GET /users/1?include=completed-attempts,attempts` fails. [completed-attempts_attempts.txt](https://github.com/cerebris/jsonapi-resources/files/152076/completed-attempts_attempts.txt)

However, as expected, `GET /users/1?include=attempts` works. [attempts.txt](https://github.com/cerebris/jsonapi-resources/files/152075/attempts.txt)

Changing the order, `GET /users/1?include=attempts,completed-attempts`, doesn't fail with a 500, however it returns the full list of `Attempts` for both the `attempts` and `completed-attempts` relationships. [attempts_completed-attempts.txt](https://github.com/cerebris/jsonapi-resources/files/152074/attempts_completed-attempts.txt)

Querying the relations directly from the AR object works correctly for both `attempts` and `completed_attempts`.

**Comment by lgebhardt** [comment]
What do your resources look like? Do you have a relationship defined for `completed_attempts`? For example:

``` ruby
class UserResource < JSONAPI::Resource
  has_many :attempts
  has_many :completed_attempts, class_name: 'Attempt'
end
```

**Comment by msbit** [comment]
Ah yes, forgot to add that.
I have both those `has_many` relations, however I hadn't added the `class_name` param, thinking it was sourced from the AR relation.
I'll check that out and see whether specifying that helps in this case.

**Comment by sdellis** [comment]
I can't even get the rails server to start with "belongs_to":
`method_missing: undefined method 'belongs_to' for OrderitemResource:Class`

Here are my "shopping cart" models:

```
class ItemResource < JSONAPI::Resource
  attributes :label, :price
  has_many :orderitems
  has_many :orders, through: :orderitems

  filter :order
end
```

```
class OrderResource < JSONAPI::Resource
  attributes :payment_fname, :payment_lname, :payment_email
  has_many :orderitems
  has_many :items, through: :orderitems
end
```

```
class OrderitemResource < JSONAPI::Resource
  attributes :qty
  belongs_to :order
  belongs_to :item 
end
```

I'm new to this gem and just testing it out to see if it's going to work for my project. Are many-to-many relationships supported?

**Comment by msbit** [comment]
@sdellis: in your situation you need to add `belongs_to` etc to the ActiveRecord model that is associated with the JSONAPI::Resource. Some of that logic is replicated in the resource layer, but most of it is defined in the model.

**Comment by sdellis** [comment]
@msbit Thank you! That worked. (But it seems that the use of underscores in properties/columns is problematic? For example, I'm getting a "param not allowed" for "payment_fname". Don't want to hijack this issue, so perhaps I should start a new one?)

**Comment by msbit** [comment]
@sdellis: Yep, there is an implicit renaming of params for the JSON API spec, so `payment_fname` becomes `payment-fname` for submissions and responses.

**Comment by lgebhardt** [comment]
The default [key formatter](https://github.com/cerebris/jsonapi-resources#key-format) is `dasherized`, which is why the params are being renamed. To use underscores set the `config.json_key_format` to `:underscored_key`

**Comment by msbit** [comment]
@lgebhardt adding `class_name` didn't change anything, and it looks like that _is_ automatically determined from the association:

`lib/jsonapi/resource.rb:778:   options[:class_name] ||= model_association.class_name`

I've discovered that I can work around it via:

```
has_many :completed_attempts, relation_name: ->(options) {
  if options[:include_directives]
    [:participations, :completed_attempts]
  else
    :completed_attempts
  end
}
```

but this seems a little brittle for my liking.

Any other suggestions for this?
